### PR TITLE
Add .tool-versions to tempdirs

### LINKED ISSUE #1392: Set `asdf local scarb...` for all temdir tests
### Which component is your bug related to?

Forge

### Foundry Version

latest

### What operating system are you using?

MacOS

### What system architecture are you using?

arm

### What happened

We should set local scarb when running tests because they are running in the temdirs so `.tool-versions` from our main directory is not enough

### Trace

_No response_

### Is there an existing issue for this?

- [ ] I have searched the existing issues and verified no issue exits for this problem.

### PR DESCRIPTION
**Pull request by drknzz**
Closes #1392 

## Introduced changes

-

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/forge/test_utils/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,12 @@
-pub mod corelib;
 pub mod runner;
 pub mod running_tests;
+
+use anyhow::Result;
+use assert_fs::fixture::PathCopy;
+
+pub fn tempdir_with_tool_versions() -> Result<assert_fs::TempDir> {
```
Comment (drknzz) [inline crates/forge/test_utils/src/lib.rs:7]: Might not be the place for it, will gladly take better ideas

File: crates/forge/src/scarb.rs [hunk 2]
```diff
@@ -75,9 +75,10 @@ mod tests {
     use indoc::{formatdoc, indoc};
     use scarb_metadata::MetadataCommand;
     use std::str::FromStr;
+    use test_utils::tempdir_with_tool_versions;
 
     fn setup_package(package_name: &str) -> TempDir {
-        let temp = TempDir::new().unwrap();
+        let temp = tempdir_with_tool_versions().unwrap();
```
Comment (war-in) [inline crates/forge/src/scarb.rs:81]: I don't think we should use functions from the `test_utils` outside the tests and `benchmarks`
cc @cptartur 

maybe it's time to consider a better place for test_utils dir?

File: crates/forge/test_utils/src/lib.rs (hunk 1 above)
Comment (war-in) [reply crates/forge/test_utils/src/lib.rs:7]: Let's put it somewhere in the `tests` dir

File: crates/forge/src/scarb.rs (hunk 2 above)
Comment (cptartur) [reply crates/forge/src/scarb.rs:81]: If `tempdir_with_tool_versions` is used in normal code, it should be added to the `forge` crate somewhere. And then we can still use it in tests.

**Review by Arcticae — APPROVED** [review]
Apart from @war-in comments it looks good to me

File: crates/forge/src/scarb.rs (hunk 2 above)
Comment (war-in) [reply crates/forge/src/scarb.rs:81]: but it shouldn't appear in the `normal` code. it is used inside some scarb tests

Comment (cptartur) [reply crates/forge/src/scarb.rs:81]: Ah, right. That's test code. I don't think it's inherently wrong to use code from `test_utils` in normal package. I'd just make sure we only use `test_utils` as a dev-dependency.

File: crates/forge/tests/e2e/collection.rs [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 use assert_fs::fixture::PathCopy;
 use indoc::indoc;
+use test_utils::tempdir_with_tool_versions;
 
 use crate::e2e::common::runner::test_runner;
 
 #[test]
 fn collection_with_lib() {
-    let temp = assert_fs::TempDir::new().unwrap();
+    let temp = tempdir_with_tool_versions().unwrap();
```
Comment (piotmag769) [inline crates/forge/tests/e2e/collection.rs:10]: We could replace those in this file with `setup_package` I think

**Review by piotmag769 — APPROVED** [review]

File: crates/forge/test_utils/src/lib.rs (hunk 1 above)
Comment (drknzz) [reply crates/forge/test_utils/src/lib.rs:7]: Kept it here, as it is used in `forge/src/scarb.rs` tests and `test_utils` is a dev-dependency anyway

**Pull request merged**